cdf/cdf.js

```diff
@@ -0,0 +1,143 @@
+const { interpolate } = require('./functions');
+
+class Cdf {
+  /**
+   * @param {number[]} xs
+   * @param {number[]} ys
+   */
+  constructor(xs, ys) {
+    this.xs = xs;
+    this.ys = ys;
+
+    if (!this.validateSize(this.xs, this.ys)) {
+      const corrected = this.convertToNewLength(this.xs.length);
+      this.xs = corrected.xs;
+      this.ys = corrected.ys;
+    }
```

Comment:
I think it should throw an error instead because we can't be sure this is correct.

Reply:
Fixed.